blog/source/_posts/VQVAE 1D.md

```diff
@@ -0,0 +1,82 @@
+tags: ['MIDI', 'ECG', 'Autoencoder', 'VQ-VAE']
+author: Samuel
+---
+
+
+## Introduction
+
+{% algrtmImgBanner VQ-VAEs/piano-hand.png piano %}
+
+VQ-VAEs are a type of generative model that merge the strengths of traditional autoencoders and vector quantization, providing a more robust and efficient method for data compression and generation. [More on VQ-VAEs](#vq-vaes-a-brief-overview)
+
+In this blog post, we'll delve into my recent research efforts centered on using VQ-VAEs for encoding 1D data—specifically, Electrocardiogram (ECG) readings and Musical Instrument Digital Interface (MIDI) data.
+
+## The Need for Data Encoding
+{% algrtmImgBanner VQ-VAEs/random-code.png code %}
+#### Importance of 1D Data
+1D data, especially in the realms of ECG and MIDI, carry crucial information. For instance, ECG data holds vital clues to a person's heart health, while MIDI sequences encode the soul of a musical piece. These data types are typically dense, capturing complex time-series information in a single dimension. As such, they are difficult to compress and store efficiently.
+
+#### Existing Solutions and Their Limitations
+Several methods are available for encoding 1D data, each with its own merits and drawbacks:
+
+- **Autoencoders:** These neural network-based techniques are generally effective for many types of data but can sometimes sacrifice details for the sake of compression.
+
+- **Fourier Transforms:** A mathematical staple for time-series data, yet not always suited for capturing non-linear or transient aspects of the data.
+
+- **Smoothing/Averaging:** Simple but potentially lossy, as they can blur important details in the data.
+
+While these methods have their uses, they also have limitations that one needs to consider depending on the application at hand. **Vector Quantized Variational Autoencoders** (VQ-VAEs) offer another avenue for exploration in the context of 1D data encoding, adding to the existing repertoire of techniques.
+
+
+## VQ-VAEs: A Brief Overview
+{% algrtmImgBanner VQ-VAEs/himeji-overview.png overview %}
+
+#### Differences from Traditional Autoencoders
+
+Traditional autoencoders consist of two primary components: an encoder that compresses input data into a lower-dimensional latent space, and a decoder that aims to reconstruct the original data from this compressed representation. Both parts are trained simultaneously with the objective of minimizing reconstruction error.
+
+However, an intriguing issue arises from this architecture: sometimes the encoder doesn't capture as much information in the latent space as one might expect. This is because decoders, especially those with high capacity, can become exceedingly proficient at "filling in the gaps," or reconstructing missing or ambiguous information. As a result, the encoder might not learn a rich or informative latent space. Instead, the decoder compensates for the encoder's shortcomings, essentially becoming too good at its job for the encoder to improve.
```

Comment:
Can you add a footnote/reference to some information demonstrating this issue?

Reply:
Turns out I mixed AEs with VAEs. Adressed this issue by talking about VAEs, they are more closely releted to VQ-VAEs either way so it turned out well 😄